### ISSUE TITLE
YOLOV4 training. while training is the aspect ration mintianed or not?

### ISSUE THREAD

**Issue by shubham-shahh** [AlexeyAB/darknet#6787]
Hi, I am training YOLOV4 with images of size 200(width)x80(height) to 600x1500. while training my .cfg file has height =416x414. is that correct or shall I make it 600x150? also if I keep 416x414, does yolo resize the images while training by keeping the aspect ratio constant? the average object size in my image is 70(height)x25(width).

thanks in advance

**Comment by stephanecharette** [comment]
> Hi, I am training YOLOV4 with images of size 200(width)x80(height) to 600x1500. while training my .cfg file has height =416x414. is that correct or shall I make it 600x150? also if I keep 416x414,

The size you've chosen and proposed to use is invalid:  https://www.ccoderun.ca/programming/2020-09-25_Darknet_FAQ/#square_network

> does yolo resize the images while training by keeping the aspect ratio constant?

The aspect ratio is ignored.  Darknet resizes the images to match the network size.

**Comment by shubham-shahh** [comment]
> > Hi, I am training YOLOV4 with images of size 200(width)x80(height) to 600x1500. while training my .cfg file has height =416x414. is that correct or shall I make it 600x150? also if I keep 416x414,
> 
> The size you've chosen and proposed to use is invalid: https://www.ccoderun.ca/programming/2020-09-25_Darknet_FAQ/#square_network
> 
> > does yolo resize the images while training by keeping the aspect ratio constant?
> 
> The aspect ratio is ignored. Darknet resizes the images to match the network size.

what should be the size according to you?

**Comment by stephanecharette** [comment]
> what should be the size according to you?

Did you click on the link?  It says:

> Choose a size that works for you and your images. The only restrictions are:
> 
> - the width has to be evenly divisible by 32
> - the height has to be evenly divisible by 32
> - you must have enough video memory to train a network of that size

**Comment by shubham-shahh** [comment]
Yes, I did saw the link, but the thing is, I'll convert the Darknet weights to tensorflow style weights and this conversion method only takes square networks
